api/response/models.py

```diff
@@ -34,3 +34,9 @@ class Response(Base, Utility):
         secondary="missing_items",
         backref=('resources'),
         lazy="joined")
+
+    @property
+    def room_response_count(self):
+        return Response.query.filter_by(
+            question_id=self.id, room_id=self.room_id
+        ).count()
```

Comment:
@RonKbS , Will this include response count to rooms that have been soft-deleted?

Reply:
@caveinn, when a room is deleted, any related tables will have the associated data removed. This is the case for hard-delete however. I think the implementation of soft delete is incomplete in this case, therefore. This is because data in associated tables is not affected by soft delete. As an example, you can take a look at the question's delete mutation. It does not account for the responses associated with a particular question.

So yes, as of now this includes responses count for archived rooms.

As a way forward a story may need to be raised that prevents queries that pick up data from related tables from including archived data as well.

Reply:
+1 to what @caveinn said, I think you should only query the Responses that have not been archived, by simply changing the way you filter it. This has been done in prior stories too.

Reply:
The responses do not have a state column, @bevkololi.

Reply:
So, do we need a story for this (responses to have a state column)?